src/main/java/com/thoughtworks/pacman/ui/screens/GameScreen.java

```diff
@@ -25,41 +55,203 @@ private GameScreen(Game game) {
+            stopTheGame = false;
+
+        }
+
+        currentFrameAt = System.currentTimeMillis();
+
+        timeDelta = currentFrameAt - lastFrameAt;
+
+        game.advance(timeDelta, stopTheGame);
 
         lastFrameAt = currentFrameAt;
+
     }
 
     public Screen getNextScreen() {
         if (game.won()) {
             return new WinScreen(game);
         } else if (game.lost() && !gamePresenter.isDying()) {
             return new LostScreen(game);
-        }
+        } else if (returnIntroScreen)
+            return new IntroScreen(game);
         return this;
     }
 
-    public void keyPressed(KeyEvent e) {
+    public void keyPressed(KeyEvent e) { // buraya escape key i ekle
         switch (e.getKeyCode()) {
-        case KeyEvent.VK_LEFT:
-            game.getPacman().setNextDirection(Direction.LEFT);
-            break;
-        case KeyEvent.VK_RIGHT:
-            game.getPacman().setNextDirection(Direction.RIGHT);
-            break;
-        case KeyEvent.VK_UP:
-            game.getPacman().setNextDirection(Direction.UP);
-            break;
-        case KeyEvent.VK_DOWN:
-            game.getPacman().setNextDirection(Direction.DOWN);
-            break;
+            case KeyEvent.VK_LEFT:
+                game.getPacman().setNextDirection(Direction.LEFT);
+                break;
+            case KeyEvent.VK_RIGHT:
+                game.getPacman().setNextDirection(Direction.RIGHT);
+                break;
+            case KeyEvent.VK_UP:
+                game.getPacman().setNextDirection(Direction.UP);
+                break;
+            case KeyEvent.VK_DOWN:
+                game.getPacman().setNextDirection(Direction.DOWN);
+                break;
+            case KeyEvent.VK_ESCAPE:
+
+                if (currentStateSETTINGS == State.SETTINGS_GONE) {
+                    game.getPacman().setNextDirection(Direction.NONE);
+                    currentStateSETTINGS = State.SETTINGS_PRESSED;
+                } else {
+                    currentStateSETTINGS = State.SETTINGS_GONE;
+                }
+
+                break;
         }
     }
+
+    private enum State {
+        HOVER_RESUME, RELEASED__RESUME, HOVER_RETURN, RELEASED__RETURN, SETTINGS_PRESSED, SETTINGS_GONE
+    }
+
+	@Override
+	public void eventDispatcher(MouseEvent event) {
+		if (event instanceof MouseEvent) {
+
+            MouseEvent e = (MouseEvent) event;
+            
+
+            if (imageClickBox.contains(e.getPoint()) && e.getID() == MouseEvent.MOUSE_CLICKED) {
+                if (currentStateSETTINGS == State.SETTINGS_GONE) {// ilk defa click ediyo settingse
+                    game.getPacman().setNextDirection(Direction.NONE);
+                    currentStateSETTINGS = State.SETTINGS_PRESSED;
+
+                } else { // click etmis ve acik menu suan
+                    currentStateSETTINGS = State.SETTINGS_GONE;
+
+                }
+            }
+            if (returnClickBox.contains(e.getPoint())) {
```

Comment:
Böyle basılan noktanın nereye düştüğü gibi bu kadar low level ara yüz işlemi yapıyoruz? Bu kodda repoda başka bir yerden mi esinlendin? Başka bir yerde de benzer bir kullanım mı var?

Reply:
Hocam, butonlarım aslında rectangle olarak tanımladığım için bu kadar low level bir kontrol yapıyorum. Başka türde kontrol bulamadım ne yazık ki.